lib_test/test_semver.ml

```diff
@@ -38,8 +38,12 @@ let test_from_parts () =
     | None -> false
     | Some _ -> true
   in
-  assert_bool "negative version rejected" (not (is_success (from_parts (-1) 0 2 [] [])));
-  assert_bool "positive version accepted" (is_success (from_parts 1 0 2 [] []))
+  assert_bool "negative version rejected"
+    (not (is_success (from_parts (-1) 0 2 [] [])));
+  assert_bool "invalid prerelease rejected"
+    (not (is_success (from_parts 1 0 0 ["0.03.7"] [])));
```

Comment:
I think this fails already because `.` is not a legal character in pre-release identifiers, and anyway, `"0.03.7"` is not a numeric identifier. I think `"03"` would be better. Test case that `"0a5b"` is a success might be a good addition.

And in this context, being complete with _correct by construction_ would mean also checking build identifiers to be composed of `[0-9A-Za-z-]`.